github-stat-tools/github-pending.py

```diff
@@ -0,0 +1,126 @@
+#! /usr/bin/python3
+
+import sys
+import subprocess
+import requests
+import json
+import pprint
+from datetime import datetime, timezone
+from optparse import OptionParser
+
+dryrun = False
+debug = False
+now = datetime.now(timezone.utc)
+
+### Helpers
+
+def search(host, q, headers):
+    search_urls = {
+        'github.com': 'https://api.github.com/search/issues'
+    }
+
+    url = search_urls[host] + '?q=' + '%20'.join(q)
+    if debug: print(f'DEBUG[search]: {url=}', file=sys.stderr)
+    res = requests.get(url, headers=headers).json()
+    if debug: print(f'DEBUG[search]: {res=}', file=sys.stderr)
+    return res
+
+### Result backends
+
+# Zabbix doesn't support much in terms of indexable <key:value>s alongside
+# the metric, like Prometheus or Loki do.  Instead, we feed all of them as
+# separate values in one input, with the hope that they all get the same
+# time stamp
+def backend_zabbix(host, basekey, type, values):
+    zabbix_command = ['zabbix_sender', '-z', 'm.openssl.org', '-i', '-']
+    zabbix_lines = [f'{host} {basekey}.{type}.{k} {v}' for k,v in values.items()]
+    zabbix_input = "\n".join(zabbix_lines) + "\n"
+    if debug or dryrun:
+        prefix = 'DEBUG[backend_zabbix]: ' if debug else ''
+        intro = 'would run this command:' if dryrun else 'running this command:'
+        for l in [ intro,
+                   '',
+                   ' '.join([ *zabbix_command, '<<_____']),
+                   *zabbix_lines,
+                   '_____',
+                   '' ]:
+            print(f'{prefix}{l}', file=sys.stderr)
+
+    if not dryrun:
+        subprocess.run(zabbix_command, input=zabbix_input, text=True);
+
+# Echoing is done in a way that's similar to Prometheus / Loki input.
+# The "metric" value is treated specially, so it becomes the actual sole
+# value, while the rest of the values are indexing label values.
+def backend_echo(host, basekey, values):
+    t = now.isoformat()
+    s = f'{basekey}' + '{' + ', '.join(
+        [ f'host="{host}"',
+          *( f'{k}="{v}"' for k,v in values.items() if k != 'metric' ) ]
+    ) + '}'
+    print(f'{t}: {s} {values["metric"]}')
+
+### Main
+
+# defaults
+host = 'github.com'
+backend = 'echo'
+# blank token is fine, but you may hit API rate limiting
+git_token = ''
+
+# parse options
+parser = OptionParser()
```

Comment:
aside: Usually I use argparse which IIRC is a bit nicer in terms of the UI it presents. The docs say optparse is deprecated. In any case it's fine.

Reply:
I'll have a look.
(I copied this bit from one of @iamamoose's scripts without looking further into it)